### PR TITLE
ENH DES pizza slices

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds functions for making pizza slices.

### CODE REVIEW

**Comment by beckermr** [comment]
OK @esheldon @rmjarvis. Some eyes on this code would be helpful. I am looking for some light code review to make sure I didn't miss anything obvious. The testing is light right now since I suspect the API will be changing. I plan to make a second PR with the final code that ties all of this together and coadds images.

File: pizza_cutter/des_pizza_cutter/_slice_data.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            yx=yx)
+
+        # now unravel
+        interp_image = interp_image.reshape(image.shape)
+        interp_noise = interp_noise.reshape(image.shape)
+
+        return interp_image, interp_noise
+    else:
+        # return a copy here since the caller expects new images
+        return image.copy(), _draw_noise_image(weight=weight, rng=rng)
```
Comment (esheldon) [inline pizza_cutter/des_pizza_cutter/_slice_data.py]: should we still use median of weight map in this case?

Comment (beckermr) [reply pizza_cutter/des_pizza_cutter/_slice_data.py]: yes good catch!

Comment (beckermr) [reply pizza_cutter/des_pizza_cutter/_slice_data.py]: Hrmmmmmm. Actually I don't think so. I think we only fill with the median when we have a bad pixel. Above in the other branch we use the reported value for any good pixels. So I think we don't need to.

Comment (esheldon) [reply pizza_cutter/des_pizza_cutter/_slice_data.py]: these areas will be treated differently if the weight map is not constant

Comment (beckermr) [reply pizza_cutter/des_pizza_cutter/_slice_data.py]: Ahhhh wait. I think I am not understanding your comment. 

For a masked pixel, do we need to muck with the weight map at all? I think we can just use it as is right? 

Right now for any pixel that needs interpolation in the image, I am replacing the weight map for that pixel by the median. I don't think we need to do that step?

Comment (beckermr) [reply pizza_cutter/des_pizza_cutter/_slice_data.py]: Note I am only replacing the weight map by its median in the masked pixels, not the full thing. So any pixels that are not masked have the same weight in either branch of the if statement.

Comment (esheldon) [reply pizza_cutter/des_pizza_cutter/_slice_data.py]: You should make sure those pixels don't have some weight map value set for them.  I don't remember if they do at times.

**Pull request merged**